src/dosext/dpmi/dnative/sigsegv.c

```diff
@@ -951,6 +958,14 @@ static void fixupsig(int sig)
 	if (sa.sa_handler == SIG_DFL || sa.sa_handler == SIG_IGN)
 		return;
 	sa.sa_flags |= SA_ONSTACK | SA_SIGINFO;
+	if (signative_block_all_sigs())
+	{
+		/* initially block all async signals. */
+		sa.sa_mask = q_mask;
+		/* SIGPROF is not in q_mask */
+		sigaddset(&sa.sa_mask, SIGPROF);
```

Comment:
I think you need to use ``setup_nf_sig()`
for that instead.

Reply:
you mean add it to `q_mask` like this:
```
diff --git a/src/arch/linux/async/signal.c b/src/arch/linux/async/signal.c
index a24053d8c..b7f8c6a8f 100644
--- a/src/arch/linux/async/signal.c
+++ b/src/arch/linux/async/signal.c
@@ -618,6 +618,7 @@ signal_pre_init(void)
   setup_nf_sig(SIG_ACQUIRE);
   setup_nf_sig(SIG_RELEASE);
   setup_nf_sig(SIGWINCH);
+  setup_nf_sig(SIGPROF);
   /* call that after all non-fatal sigs set up */
 #ifdef __i386__
   newsetsig(SIGILL, minfault);
```
?

Reply:
in that case SIGPROF is no longer special, so it'll be fixed up in the loop like the other async signals

Reply:
So what is the problem if it is no longer
special?

Reply:
Of course I intended to not have it special-cased.